lib/index.js

```diff
@@ -54,12 +65,20 @@ class EvervaultClient {
       return originalRequest.apply(this, args);
     }
 
-    const origCreateSecureContext = tls.createSecureContext;
-    tls.createSecureContext = (options) => {
-      const context = origCreateSecureContext(options);
-      context.context.addCACert(pem);
-      return context;
-    };
+    let attempts = 0
+    https.request = function (...args) {
+      if (isIgnoreRequest(args)) return originalRequest.apply(this, args);
+
+      if (!pem && typeof getCertPromise === 'promise' && attempts < 3) {
```

Comment:
AFAIK `typeof getCertPromise === 'promise'` will always return `Object` as `Promise` prototype is `Object`. This should be `getCertPromise instanceOf Promise`.